op-node/rollup/derive/engine_queue.go

```diff
@@ -457,21 +486,89 @@ func (eq *EngineQueue) forceNextSafeAttributes(ctx context.Context) error {
 			return NewCriticalError(fmt.Errorf("unknown InsertHeadBlock error type %d: %w", errType, err))
 		}
 	}
+	eq.safeAttributes = eq.safeAttributes[1:]
+	eq.logSyncProgress("processed safe block derived from L1")
+
+	return nil
+}
+
```

Comment:
do you mind adding godocs to these new functions?

Reply:
i see that there are godocs on the interface, any implementation specific details that would be useful in the godoc? perhaps the todo

Reply:
I don't like duplicating docs, and the interface is the first place the user looks & thus the best to document, so I didn't add docs here. No inspiration for implementation-specific notes either. Do you have suggestions for what to document?